### ISSUE TITLE
Graph Queries - recursive EdgeQL 

### ISSUE THREAD

**Issue by kerimcharfi** [edgedb/edgedb#4168]
edgedb names it self a graph-relational database. However, I miss basic graph query abilities in edgedb.
Graph queries can be realisied in SQL by using a recursive query in SQL. 

PostgreSQL Graph Search Practices - 10 Billion-Scale Graph with Millisecond Response
https://www.alibabacloud.com/blog/postgresql-graph-search-practices---10-billion-scale-graph-with-millisecond-response_595039

I can't find something similar in EdgeQl. Am I missing something ?
An alternative solution could be to be able to overwrite variables inside an EdgeQL `for `statement which are declared outside of the "loop". This way one could remember the current graph frontier and expand it in every iteration.

**Comment by elprans** [comment]
Right, support for transitive closure (a.k.a recursive CTE in SQL) is something we are planning to implement in EdgeQL soon.  See also an earlier discussion in #2634.
